### PR TITLE
mail: add Message-Id to mail headers

### PR DESCRIPTION
**Pull request by FredClement91**
##### SUMMARY
<!— Your description here –>
Add Message-Id to the headers to avoid servers to see our mails as spams

##### SUMMARY

We have generated and added the message-ID field that is mandatory for some mail servers

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module/Plugin Pull Request
- Refactoring Pull Request
- Test Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/mail.py [hunk 1]
```diff
@@ -246,6 +246,7 @@ def main():
             subtype=dict(type='str', default='plain', choices=['html', 'plain']),
             secure=dict(type='str', default='try', choices=['always', 'never', 'starttls', 'try']),
             timeout=dict(type='int', default=20),
+            message_id=dict(type='str', aliases=['Message-ID'])
```
Comment (felixfontein) [inline plugins/modules/mail.py:249]: Please use `message_id` without an alias.
```suggestion
            message_id=dict(type='str'),
```
Also you need to document this new option in `DOCUMENTATION` as the top of the file.

File: plugins/modules/mail.py [hunk 2]
```diff
@@ -341,6 +343,8 @@ def main():
     msg['From'] = formataddr((sender_phrase, sender_addr))
     msg['Date'] = formatdate(localtime=True)
     msg['Subject'] = Header(subject, charset)
+    if message_id:
+      msg['Message-ID'] = make_msgid(domain=message_id)
```
Comment (felixfontein) [inline plugins/modules/mail.py:347]: What is `make_msgid`?

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment). Thanks.

File: plugins/modules/mail.py (hunk 2 above)
Comment (fred-clement-91) [reply plugins/modules/mail.py:347]: email.utils.make_msgid(idstring=None, domain=None)[¶](https://docs.python.org/3/library/email.utils.html#email.utils.make_msgid)

    Returns a string suitable for an [RFC 2822](https://datatracker.ietf.org/doc/html/rfc2822.html)-compliant Message-ID header. Optional idstring if given, is a string used to strengthen the uniqueness of the message id. Optional domain if given provides the portion of the msgid after the ‘@’. The default is the local hostname. It is not normally necessary to override this default, but may be useful certain cases, such as a constructing distributed system that uses a consistent domain name across multiple hosts.

Comment (felixfontein) [reply plugins/modules/mail.py:347]: In that case you need to import it. Right now this is an undefined symbol. (See also the failing tests.)

**Comment by felixfontein** [comment]
ping @FredClement91 

needs_info

**Pull request closed without merging**